zipline/utils/cache.py

```diff
@@ -298,9 +304,25 @@ def __enter__(self):
         return self
 
     def __exit__(self, *exc_info):
+        self._tmpfile.__exit__(*exc_info)
         if exc_info[0] is None:
             self._commit()
-        self._tmpfile.__exit__(*exc_info)
+
+    @property
```

Comment:
also, do we ever need to access the tmpfile directly, we forward all attrs to the tmpfile through our `__getattribute__`, so `working_file.close` is the same as `workingfile.tmpfile.close`